### PR TITLE
feat: add `array/base/mskbinary3d`

### LINKED ISSUE #3178: [RFC]: add `array/base/mskbinary3d`
### Description

This RFC proposes adding a new package to the `array/base/*` namespace: `@stdlib/array/base/mskbinary3d`.

This package is the 3d equivalent of [`@stdlib/array/base/mskbinary2d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/mskbinary2d).

To see an example of a similar package, see [`@stdlib/array/base/mskunary3d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/mskunary3d).

While the package proposed in this RFC will **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by rylany27**
## Description

> What is the purpose of this pull request?

This pull request:

- implements the mskbinary3d package for masked binary operations in the array/base namespace.
- and provides a 3D equivalent of the mskbinary2d package for binary callbacks using a mask array.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3178

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/array/base/mskbinary3d/package.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+    "name": "@stdlib/array/base/mskbinary3d",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/mskbinary3d/package.json]: The indentation in this file is off. Looks like it just needs to be de-dented by 2 spaces.

**Review by kgryte — COMMENTED** [review]
Thank you for working on this, @rylany27. Overall, this PR is looking good. After the `package.json` file indentation issue is resolved (and it needs a trailing newline), I'll take a final closer look.

**Comment by rylany27** [comment]
Thank you for the review, @kgryte! I appreciate your feedback. Should I go ahead and make another commit to resolve the package.json indentation issue and add the trailing newline?

**Comment by kgryte** [comment]
The test coverage CI failure seems to be an issue with the GitHub workflow, not this PR.

@rylany27 Yes, please do.

**Comment by kgryte** [comment]
@rylany27 I just realized that the package you've added is missing a README. Would you mind adding to this PR?

**Review by kgryte — COMMENTED** [review]
Once the missing README is added, this PR is ready for merge.

**Comment by rylany27** [comment]
Hi @kgryte , I've added the README. Let me know if there’s anything else needed. Thanks!

File: lib/node_modules/@stdlib/array/base/mskbinary3d/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# mskbinary3d
+
+> Apply a binary callback to elements in two three-dimensional nested input arrays according to elements in a three-dimensional nested
+> mask array and assign results to elements in a three-dimensional nested output array.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/mskbinary3d/README.md]:
```suggestion
> Apply a binary callback to elements in two three-dimensional nested input arrays according to elements in a three-dimensional nested mask array and assign results to elements in a three-dimensional nested output array.
```

File: lib/node_modules/@stdlib/array/base/mskbinary3d/README.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+## Usage
+
+```javascript
+var mskbinary3d = require( '@stdlib/array/base/mskbinary3d' );
+```
+
+#### mskbinary3d( arrays, shape, fcn )
+
+Applies a binary callback to elements in two three-dimensional nested input arrays according to elements in a three-dimensional nested
+mask array and assigns results to elements in a three-dimensional nested output array.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/mskbinary3d/README.md]:
```suggestion
Applies a binary callback to elements in two three-dimensional nested input arrays according to elements in a three-dimensional nested mask array and assigns results to elements in a three-dimensional nested output array.
```

File: lib/node_modules/@stdlib/array/base/mskbinary3d/README.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+var x = [
+    [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ],
+    [ [ 5.0, 6.0 ], [ 7.0, 8.0 ] ]
+];
+var z = zeros3d( [ 2, 2, 2 ] );
+
+var mask = [
+    [ [ 0.0, 1.0 ], [ 0.0, 0.0 ] ],
+    [ [ 1.0, 0.0 ], [ 0.0, 1.0 ] ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/mskbinary3d/README.md]:
```suggestion
    [ [ 0, 1 ], [ 0, 0 ] ],
    [ [ 1, 0 ], [ 0, 1 ] ]
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @rylany27!

**Pull request merged**